spark-board-ui/src/useDagAnimation/useDagAnimation.js

```diff
@@ -13,12 +13,17 @@ import { stepSimulation } from './physics';
  * @param {Array} nodes react-flow nodes.
  * @param {Array} edges react-flow edges.
  * @param {Map} targetPositions Map from the node IDs to the target positions (objects with `x` and `y`).
- * @param {Callback} setNodes react-flow state setter to update the nodes.
+ * @param {Callback} setCurrentPositions react-flow state setter to update the node positions.
  * @returns Two elements:
  *          1. Callback that will start the animation when called.
  *          2. Callback that will update the node positions with the ones given.
  */
 export function useDagAnimation(nodes, edges, targetPositions, setNodes) {
+    // positions where the nodes are being pulled to (if the animation is enabled)
+    const [currentPositions, setCurrentPositions] = useState(new Map(
+        nodes.map(node => [node.id, node.position])
+    ));
+
```

Comment:
do you think it makes sense to make this a state?

it's not being shared anywhere outside of this hook and only updated internally. It feels like it's an unnecesary middleman.

Reply:
I liked this because the effect would be the "node updater" and, if we add another state, we could just add a new dependency to the effect... But yes, maybe it's useless right now and it would be a good change in case we add that hypotetical new state in the future